### ISSUE TITLE
Current state of the go implementation and roadmap

### ISSUE THREAD

**Issue by Metamogul** [pact-foundation/pact-go#406]
## Expected behaviour

When trying to find information on the projects feature set and roadmap, I'd expect to find information that enables me to propose a decision wether and when to implement pact contract tests across the different services of our product.

## Actual behaviour

The Roadmap section of the README currently links two pages: The [roadmap on pact.io](https://docs.pact.io/roadmap) last edited ~4 years ago and [a roadmap document in this projects GitHub wiki](https://github.com/pact-foundation/pact-go/wiki/Native-implementation-roadmap), last edited ~8 years ago. Obviously this cannot help gaining insights.

Trying to join the projects slack as pointed out in the README when looking for help leads to a company-private workspace that is not open to the general public. Following the help section on pact.io also does not provide any other sources of information.

## Steps to reproduce

Read the README. Try to join slack.

## Follow-up questions

The issues seems to indicate a lack of community resources to drive the project; outdated and scattered documentation (e.g. outdated docs on the go implementation on pact.io) or the removal of GraphQL support in the 2.x.x version of the go module support that impression, while the communication on the whole pact framework seems to have shifted to a company-private approach. This contributes to a situation where there is little transparency over the state of the project, its maintainers and their plans, and leaves me with doubts whether I should recommend the implementation of pact contract tests in our product. Adopting an only partly open source project as well as contributing seems risky long term.

Is there still an open community space in which I could get in touch with the maintainers? Are there any notable public contributions to this project at the moment or plans to keep it public?

**Comment by mefellows** [comment]
Ah, the inviter looks to have stopped working.

Use this for now:
https://join.slack.com/t/pact-foundation/shared_invite/zt-2h7ovnj5x-VD82RSfK68jseNXTQ_KTdA

**Comment by YOU54F** [comment]
Sorry about the slack issue @Metamogul. we've had issues recently due to the way Slack works, it was tracked in the below issue.

https://github.com/pact-foundation/devrel/issues/26

tl;dr the invite link only lasts for a limited amount of users and we need to refresh the link, and then update the redirect for our invite link.


>  (e.g. outdated docs on the go implementation on pact.io) 

ahh, 

https://docs.pact.io/implementation_guides/go/readme

It seems that when we moved over to v2, we have lost our sync docs job

https://github.com/pact-foundation/pact-go/blob/v1.x.x/.github/workflows/trigger_pact_docs_update.yml

we will want to restore that against the `master` branch, which will allow it to trigger the sync job in the docs repo

https://github.com/pact-foundation/docs.pact.io/blob/master/scripts/sync/pact_go.rb

We would be happy to accept a pull request, or for it to be raised as a separate issue to track and resolve. It's obviously less than desirable for our centralised docs to be out of date with the readme in github, especially as we want to increase adoption of pact-go v2 which invariably teases out bugs.

> while the communication on the whole pact framework seems to have shifted to a company-private approach. 

outside of the slack invite being locked down (we don't have a paid slack account ), how would you suggest this is the case?

The framework is as open-source as it has ever been, we do more companies using the project, and some of those customers choose PactFlow (which is a commercial product), however the Pact ecosystem which serves both Pact Open source and PactFlow commercial users.

From a maintainer perspective, we are vastly outweighed by the number of users, vs the number of maintainers and active contributors. We all have varying pressures at work, and in our home life, but we are as committed as always to Pact being open and accessible for everyone


>This contributes to a situation where there is little transparency over the state of the project, its maintainers and their plans, and leaves me with doubts whether I should recommend the implementation of pact contract tests in our product. Adopting an only partly open source project as well as contributing seems risky long term.

- Monthly blog https://docs.pact.io/blog
- Pact Go Roadmap https://github.com/pact-foundation/pact-go/projects/2 
- Pact Feature Request board https://pact.canny.io/

> Is there still an open community space in which I could get in touch with the maintainers? Are there any notable public contributions to this project at the moment or plans to keep it public?

There are no plans to make it not public... and you are making contact in an open community space here in github.

You can see public contributions via git commit history, and pull requests. They ebb and flow, with users wants and needs. 

There has been several go users, who have dived into the Pact matrix, learned about Pact's shared core (built in Rust) and done some Polyglot programming in the process.

**Comment by YOU54F** [comment]
Slack is locked down and not-open source, we lose history of conversations after 90 days. It's a shame to lose the valuable insight from the community, and not be able to serve it for others, so we did something about it.

We do so by leveraging another oss product https://github.com/Linen-dev/linen.dev where our slack history is fully searchable and hosted here

https://www.linen.dev/s/pact-foundation

That allows us to preserve conversation that takes place on Slack, for users for years to come. 

We were previously scraping it and hosting it here https://docs.pact.io/slack but it wasn't as nice to consume

**Comment by YOU54F** [comment]
If you have ideas about pact-go's future, and want to help make it awesome, we are all ears, both in term so ideas, and PR's!

**Comment by YOU54F** [comment]
Sync job has been restored, and the website has been updated with the latest documentation

https://docs.pact.io/implementation_guides/go

thanks for bringing it to our attention

**Comment by Metamogul** [comment]
Hey @YOU54F, thx for restoring the sync job and the various clarifications & transparency! This helps me assess the situation much better 😌 When it comes to contributions I'm already overbooked myself with my time – so my ability to contribute to the project depends largely on my employer's allocation of my hours. If I'll be given more time on our Pact implementation and I'll need to amend to the project I'll be pushing to make these publicly available as PR. For now, I think this issue can be closed. Thanks again for your time and the insightful posts.
